noir-projects/aztec-nr/aztec/src/context/private_context.nr

```diff
@@ -66,7 +67,7 @@ pub struct PrivateContext {
     nullifiers: BoundedVec<Nullifier, MAX_NULLIFIERS_PER_CALL>,
 
     private_call_requests: BoundedVec<PrivateCallRequest, MAX_PRIVATE_CALL_STACK_LENGTH_PER_CALL>,
-    public_call_requests: BoundedVec<PublicCallRequest, MAX_PUBLIC_CALL_STACK_LENGTH_PER_CALL>,
+    public_call_requests: BoundedVec<Counted<PublicCallRequest>, MAX_ENQUEUED_CALLS_PER_CALL>,
```

Comment:
Remove `counter` from `PublicCallRequest` and use `Counted` to add `counter` to any struct.

Will do the same for other side effects in another PR.